### ISSUE TITLE
[MDEP-561] Align base/unique default behavior between dependency/jar/assembly plugins

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#995]
**[Marcin Domanski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=domaru)** opened **[MDEP-561](https://issues.apache.org/jira/browse/MDEP-561?redirect=false)** and commented

Currently dependency plugin in `copy-dependencies` goal has an `useBaseVersion` option that defaults to true to force it to us unique versions for snapshots.

This default behavior is different to default behavior of jar/archiver plugin where you have `useUniqueVersions` option that defaults to true
Similarly Assembly plugin also by default uses unique versions that you can override by outputFileNameMapping element.

To sum up, it should be more user friendly for all ASF Maven plugins to share the same default settings regarding unique/base versioning. Otherwise ASF plugins would not produce self compatible output  by default.

**Example:** If you use jar plugin to create jar manifest with classpath and dependency plugin to provide those classpath dependencies classpath dependency names will not match the ones copied by the dependency plugin unless you explicitly disable useBaseVersions

---

**Affects:** 3.0.0
